### PR TITLE
Use route model binding

### PR DESCRIPTION
**Pull request by Riari**
This adds route model binding as per #40.

### CODE REVIEW

File: src/ForumServiceProvider.php [hunk 1]
```diff
@@ -28,6 +32,12 @@ public function boot()
         $this->registerPackageNamespaces();
 
         View::composer('*', 'Bitporch\Forum\ViewComposers\GroupComposer');
+
+        Route::model('discussion', Discussion::class);
+        Route::bind('group', function ($value) {
+            return Group::where('slug', $slug)->firstOrFail();
```
Comment (AndreasElia) [inline src/ForumServiceProvider.php]: is `$value` meant to be `$slug`? 😄

Comment (Riari) [inline src/ForumServiceProvider.php]: Oops 😛

File: src/ForumServiceProvider.php [hunk 2]
```diff
@@ -28,6 +32,12 @@ public function boot()
         $this->registerPackageNamespaces();
 
         View::composer('*', 'Bitporch\Forum\ViewComposers\GroupComposer');
+
+        Route::model('discussion', Discussion::class);
```
Comment (deleugpn) [inline src/ForumServiceProvider.php]: This is not necessary when 
- The name of the variable is the same as the model;
- Using default primary key;

File: src/ForumServiceProvider.php [hunk 3]
```diff
@@ -28,6 +32,12 @@ public function boot()
...
+        Route::bind('group', function ($slug) {
+            return Group::where('slug', $slug)->firstOrFail();
```
Comment (deleugpn) [inline src/ForumServiceProvider.php]: this one is necessary because we're using a slug.

File: src/ForumServiceProvider.php (hunk 2 above)
Comment (Riari) [inline src/ForumServiceProvider.php]: I know, but I prefer to be at least somewhat explicit here. Not keen on some of Laravel's magic - there's already enough going on even with this. Can get rid of it though if others are in favour of that.

Comment (deleugpn) [inline src/ForumServiceProvider.php]: I already got rid of it on #84 but I totally see your point. Should be brought into the table and discussed whether to bring back or not.

**Pull request merged**